app/main.py

```diff
@@ -4,3 +1,47 @@
+import os
+import requests
+from requests.exceptions import (
+    ConnectionError,
+    JSONDecodeError,
+    RequestException
+)
+
+
+URL = "http://api.weatherapi.com/v1/current.json?"
+API_KEY = os.environ["API_KEY"]
+FILTERING = os.environ.get("FILTERING", "Paris")
+
+
+def get_weather() -> dict:
+    url = f"{URL}q={FILTERING}&key={API_KEY}"
+    weather_data = fetch_data(url)
+    print_weather_data(weather_data)
+
+
+def fetch_data(url: str) -> dict:
+    try:
+        response = requests.get(url)
+        response.raise_for_status()
+        data = response.json()
+        return data
```

Comment:
The function `fetch_data` should return an empty dictionary or `None` in the exception handling blocks to avoid returning `None` implicitly, which could lead to errors when `print_weather_data` is called.